lib/tasks/multitenant_db_merge.rake

```diff
@@ -0,0 +1,344 @@
+
+  # Port users
+  user_map = -> (x) {
+    x.except('id', 'fund_id', 'line', 'created_at', 'updated_at')
+     .merge({
+       'fund_id' => @fund_id,
+       'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+       'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York")
+     })
+  }
+  @user_mappings = easy_mass_insert User, 'users', user_map
+
+  # Port patients
+  patient_map = -> (x) {
+    x.except('id', 'fund_id', 'created_at', 'updated_at', 'line_id', 'clinic_id', 'pledge_generated_by_id', 'pledge_sent_by_id', 'last_edited_by_id')
+     .merge({
+       'fund_id' => @fund_id,
+       'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+       'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+       'clinic_id' => @clinic_mappings[x['clinic_id']],
+       'line_id' => @line_mappings[x['line_id']],
+       'pledge_generated_by_id' => @user_mappings[x['pledge_generated_by_id']],
+       'pledge_sent_by_id' => @user_mappings[x['pledge_sent_by_id']],
+       'last_edited_by_id' => @user_mappings[x['last_edited_by_id']]
+     })
+  }
+  @patient_mappings = easy_mass_insert Patient, 'patients', patient_map
+
+  # Port archived patients
+  archived_patient_map = -> (x) {
+    x.except('id', 'fund_id', 'created_at', 'updated_at', 'line_id', 'clinic_id', 'pledge_generated_by_id', 'pledge_sent_by_id', 'last_edited_by_id')
+     .merge({
+       'fund_id' => @fund_id,
+       'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+       'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+       'clinic_id' => @clinic_mappings[x['clinic_id']],
+       'line_id' => @line_mappings[x['line_id']],
+       'pledge_generated_by_id' => @user_mappings[x['pledge_generated_by_id']],
+       'pledge_sent_by_id' => @user_mappings[x['pledge_sent_by_id']],
+     })
+  }
+  @archived_patient_mappings = easy_mass_insert ArchivedPatient, 'archived_patients', archived_patient_map
+
+  # Note about subobjects: there are some abandoned records (e.g. from patients who were deleted)
+  # so we don't count check on these
+  # Port calls
+  call_map = -> (x) {
+    res = x.except('id', 'can_call_id', 'fund_id', 'created_at', 'updated_at')
+      .merge({
+        'fund_id' => @fund_id,
+        'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+        'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+        'can_call_id' => if x['can_call_type'] == 'Patient'
+                           @patient_mappings[x['can_call_id']]
+                         elsif x['can_call_type'] == 'ArchivedPatient'
+                           @archived_patient_mappings[x['can_call_id']]
+                         else
+                           raise "unexpected type - row #{x}"
+                         end
+      })
+    res['can_call_id'].nil? ? nil : res
```

Comment:
What's this line doing? Is it dropping abandoned subobjects by having the lamda return nil?

Reply:
yeah, I encountered a few cases of exactly this - abandoned subobjects. This guards against that case.